pkg/sdn/plugin/proxy.go

```diff
@@ -14,12 +14,23 @@ import (
 	"k8s.io/kubernetes/pkg/client/cache"
 	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
 	pconfig "k8s.io/kubernetes/pkg/proxy/config"
+	ktypes "k8s.io/kubernetes/pkg/types"
 	utilwait "k8s.io/kubernetes/pkg/util/wait"
 )
 
+type firewallItem struct {
+	ruleType osapi.EgressNetworkPolicyRuleType
+	net      *net.IPNet
+}
+
+type policyFirewallItem struct {
+	uid      ktypes.UID
+	firewall []firewallItem
+}
+
 type proxyFirewallItem struct {
-	policy osapi.EgressNetworkPolicyRuleType
-	net    *net.IPNet
+	namespaceFirewalls []policyFirewallItem
```

Comment:
doing the map looks plausible and would make the code more readable, so +1 from me if it works